src/runtime/storageNG/handle.ts

```diff
@@ -118,22 +116,60 @@ export abstract class Handle<T extends CRDTTypeRecord> {
   }
 }
 
+abstract class PreEntityMutationHandle<T extends CRDTTypeRecord> extends Handle<T> {
```

Comment:
Could you add a comment explaining what this class is for? Particularly addressing the meaning of "pre-mutation".